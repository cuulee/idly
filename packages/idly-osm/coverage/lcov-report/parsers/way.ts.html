<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for parsers/way.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">parsers</a> way.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { weakCache } from 'idly-common/lib/misc/weakCache';
import {
  EntityTable,
  OsmGeometry,
  Way,
  EntityType
} from 'idly-common/lib/osm/structures';
<span class="branch-0 cbranch-no" title="branch not covered" ></span>
import { isArea } from '../helpers/isArea';
import { tagClassesPrimary } from '../tagClasses/tagClasses';
import { presetsMatcherCached } from '../presets/presets';
&nbsp;
const tagClassesPrimaryCache = weakCache(tagClassesPrimary);
&nbsp;
export function wayPropertiesGen(way: Way, table: EntityTable) {
  const geometry = isArea(way) ? OsmGeometry.AREA : OsmGeometry.LINE;
  const [tagsClass, tagsClassType] = tagClassesPrimaryCache(way.tags);
  const match = presetsMatcherCached(geometry)(way.tags);
  return {
    name: way.tags.get('name') || way.tags.get('ref'),
    icon: match &amp;&amp; match.icon,
    geometry,
    tagsClass,
    tagsClassType
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 29 2017 19:59:11 GMT+0530 (IST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
