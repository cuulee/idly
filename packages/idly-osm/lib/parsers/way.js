"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lib_1 = require("idly-common/lib");
const isArea_1 = require("../helpers/isArea");
const tagClasses_1 = require("../tagClasses/tagClasses");
const presets_1 = require("../presets/presets");
const tagClassesPrimaryCache = lib_1.weakCache(tagClasses_1.tagClassesPrimary);
function wayPropertiesGen(way, table) {
    const geometry = isArea_1.isArea(way) ? lib_1.OsmGeometry.AREA : lib_1.OsmGeometry.LINE;
    const [tagsClass, tagsClassType] = tagClassesPrimaryCache(way.tags);
    const match = presets_1.presetsMatcherCached(geometry)(way.tags);
    return {
        name: way.tags.get('name') || way.tags.get('ref'),
        icon: match && match.icon,
        geometry,
        tagsClass,
        tagsClassType
    };
}
exports.wayPropertiesGen = wayPropertiesGen;
//# sourceMappingURL=way.js.map