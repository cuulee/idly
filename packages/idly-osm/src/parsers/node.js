"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var presetsInit_1 = require("./presetsInit");
var weakCache_1 = require("idly-common/lib/misc/weakCache");
var getNodeGeometry_1 = require("../helpers/getNodeGeometry");
exports.DEFAULT_NODE_ICON = 'circle';
exports.nodePropertiesGen = weakCache_1.weakCache2(function (node, parentWay) {
    return exports.applyNodeMarkup(getNodeGeometry_1.getNodeGeometry(node.id, parentWay), node.tags);
});
exports.applyNodeMarkup = function (geometry, tags) {
    var match = presetsInit_1.presetMatch(tags, geometry); //presetsMatcherCached(geometry)(tags);
    return {
        icon: (match && match.icon) || exports.DEFAULT_NODE_ICON,
        name: tags['name'],
        geometry: geometry
    };
};
