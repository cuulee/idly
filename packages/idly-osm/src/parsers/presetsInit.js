"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var presetIndex_1 = require("idly-data/lib/presets/presetIndex");
exports.all = presetIndex_1.presetIndex().init();
exports.presetMatch = exports.all.match;
