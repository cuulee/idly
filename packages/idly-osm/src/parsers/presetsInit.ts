import { OsmGeometry, Tags } from 'idly-common/lib/osm/structures';
import { presetIndex } from 'idly-data/lib/presets/presetIndex';

export const all = presetIndex().init();

export const presetMatch: (tags: Tags, geometry: OsmGeometry) => any =
  all.match;
