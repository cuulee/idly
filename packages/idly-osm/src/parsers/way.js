"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var presetsInit_1 = require("./presetsInit");
var weakCache_1 = require("idly-common/lib/misc/weakCache");
var structures_1 = require("idly-common/lib/osm/structures");
var isArea_1 = require("../helpers/isArea");
var tagClasses_1 = require("../tagClasses/tagClasses");
var tagClassesPrimaryCache = weakCache_1.weakCache(tagClasses_1.tagClassesPrimary);
exports.wayPropertiesGen = weakCache_1.weakCache(function (way) {
    var geometry = isArea_1.isArea(way) ? structures_1.OsmGeometry.AREA : structures_1.OsmGeometry.LINE;
    var _a = tagClassesPrimaryCache(way.tags), tagsClass = _a[0], tagsClassType = _a[1];
    var match = presetsInit_1.presetMatch(way.tags, geometry); //presetsMatcherCached(geometry)(way.tags);
    return {
        name: way.tags['name'] || way.tags['ref'],
        icon: match && match.icon,
        geometry: geometry,
        tagsClass: tagsClass,
        tagsClassType: tagsClassType
    };
});
