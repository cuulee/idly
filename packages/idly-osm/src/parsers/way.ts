import { EntityTable, OsmGeometry, Way, weakCache } from 'idly-common/lib';

import { isArea } from '../helpers/isArea';
import { tagClassesPrimary } from '../tagClasses/tagClasses';
import { presetsMatcherCached } from '../presets/presets';

const tagClassesPrimaryCache = weakCache(tagClassesPrimary);

export function wayPropertiesGen(way: Way, table: EntityTable) {
  const geometry = isArea(way) ? OsmGeometry.AREA : OsmGeometry.LINE;
  const [tagsClass, tagsClassType] = tagClassesPrimaryCache(way.tags);
  const match = presetsMatcherCached(geometry)(way.tags);
  return {
    name: way.tags.get('name') || way.tags.get('ref'),
    icon: match && match.icon,
    geometry,
    tagsClass,
    tagsClassType
  };
}
