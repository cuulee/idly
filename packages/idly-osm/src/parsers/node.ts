import { presetMatch } from './presetsInit';
import { weakCache2 } from 'idly-common/lib/misc/weakCache';
import {
  Node,
  EntityType,
  NodeGeometry,
  ParentWays,
  Tags,
  OsmGeometry
} from 'idly-common/lib/osm/structures';
import { getNodeGeometry } from '../helpers/getNodeGeometry';

import { Set as ImSet } from 'immutable';

export const DEFAULT_NODE_ICON = 'circle';

export const nodePropertiesGen = weakCache2(
  (node: Node, parentWay: ImSet<string>) => {
    return applyNodeMarkup(getNodeGeometry(node.id, parentWay), node.tags);
  }
);

export const applyNodeMarkup = (geometry: NodeGeometry, tags: Tags) => {
  const match = presetMatch(tags, geometry); //presetsMatcherCached(geometry)(tags);
  return {
    icon: (match && match.icon) || DEFAULT_NODE_ICON,
    name: tags['name'],
    geometry
  };
};
