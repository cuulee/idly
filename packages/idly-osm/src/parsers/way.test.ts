import {
  entityTableGen,
  genLngLat,
  nodeFactory,
  tagsFactory,
  wayFactory
} from 'idly-common/lib';

import { wayPropertiesGen } from '../parsers/way';

describe('wayCombiner', () => {
  const n1 = nodeFactory({ id: 'n1', loc: genLngLat([1, 2]) });
  const n2 = nodeFactory({ id: 'n2', loc: genLngLat([3, 4]) });
  const n3 = nodeFactory({ id: 'n3', loc: genLngLat([1, 4]) });
  const node = nodeFactory({ id: 'n1' });
  const way = wayFactory({
    id: 'w1',
    nodes: ['n1'],
    tags: tagsFactory({ highway: 'residential' })
  });

  const g = entityTableGen(new Map(), [n1, n2, n3]);
  it('should behave...', () => {
    const result = wayPropertiesGen(way, g);
    expect(result).toMatchSnapshot();
  });

  it('should take name', () => {
    const w2 = wayFactory({
      id: 'w1',
      nodes: ['n1'],
      tags: tagsFactory({ highway: 'residential', name: 'great highway' })
    });
    const gg = entityTableGen(new Map(), [node, w2]);
    const result = wayPropertiesGen(w2, gg);
    expect(result.name).toEqual('great highway');
  });
});
