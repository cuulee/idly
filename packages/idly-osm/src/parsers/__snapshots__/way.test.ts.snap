// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wayCombiner should behave... 1`] = `
Object {
  "geometry": "line",
  "icon": "highway-residential",
  "name": undefined,
  "tagsClass": "tag-highway",
  "tagsClassType": "tag-highway-residential",
}
`;
